lib/urbanopt/geojson/geo_file.rb

```diff
@@ -119,36 +161,100 @@ def get_feature_by_id(feature_id)
       end
 
-      ##
-      # Returns the +geojson_schema+ .
-      def schema
+      def self.get_geojson_schema(strict)
+        result = nil
         if @@geojson_schema.nil?
           @@schema_file_lock.synchronize do
-            File.open(schema_file, 'r') do |file|
-              @@geojson_schema = JSON.parse(file.read, symbolize_names: true)
+            File.open(File.dirname(__FILE__) + '/schema/geojson_schema.json') do |f|
+              result = JSON.parse(f.read, symbolize_names: true)
             end
           end
         end
+        return result
+      end
 
-        return @@geojson_schema
+      def self.get_building_schema(strict)
+        result = nil
+        File.open(File.dirname(__FILE__) + '/schema/building_properties.json') do |f|
+          result = JSON.parse(f.read)
+        end
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result
+      end
+      
+      def self.get_district_system_schema(strict)
+        result = nil
+        File.open(File.dirname(__FILE__) + '/schema/district_system_properties.json') do |f|
+          result = JSON.parse(f.read)
+        end
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result
+      end
+      
+      def self.get_region_schema(strict)
+        result = nil
+        File.open(File.dirname(__FILE__) + '/schema/region_properties.json') do |f|
+          result = JSON.parse(f.read)
+        end
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result
       end
 
-      ##
-      # Validates the GeoJSON file against the schema.
-      def valid?
-        return JSON::Validator.validate(schema, @geojson)
+      def self.get_electrical_connector_schema(strict)
+        result = nil
+        File.open(File.dirname(__FILE__) + '/schema/electrical_connector_properties.json') do |f|
+          result = JSON.parse(f.read)
+        end
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result
       end
 
-      ##
-      # Returns detailed validation results.
-      def validation_errors
-        return JSON::Validator.fully_validate(schema, @geojson)
+      def self.get_electrical_junction_schema(strict)
+        result = nil
+        File.open(File.dirname(__FILE__) + '/schema/electrical_junction_properties.json') do |f|
+          result = JSON.parse(f.read)
+        end
+        if strict
+          result['additionalProperties'] = true
+        else
+          result['additionalProperties'] = false
+        end
+        return result      
       end
+
+      strict = true
+      @@geojson_schema = get_geojson_schema(strict)
```

Comment:
there are also thermal connectors and thermal junctions...should you add/validate those here too?